cpp/src/arrow/array/concatenate.cc

```diff
@@ -169,22 +169,26 @@ Status PutOffsets(const std::shared_ptr<Buffer>& src, Offset first_offset, Offse
   return Status::OK();
 }
 
+// Implement of `arrow::Concatenate`.
+// N.B. The ConcatenateImpl should accept a reference to a vector of ArrayData,
+// instead of a temporary value.
```

Comment:
What does it mean? Is it a TODO? Otherwise, I don't see a reason for this comment.

Reply:
Well not a todo, just tell the user not to use like:

```c++
ConcatenateImpl i{/* get prvalue */GenerateArray()};
i.concate()
```

That would be unsafe, because `ConcatenateImpl` holds a const reference. If you think it's not neccessary, I'll remove it.

Reply:
This is a private API, so I don't think it's necessary.

Reply:
Fixed now